packages/datadog-plugin-aws-sdk/src/services/eventbridge.js

```diff
@@ -26,19 +26,32 @@ class EventBridge {
     if (operation === 'putEvents' &&
       request.params &&
       request.params.Entries &&
-      request.params.Entries.length > 0 &&
-      request.params.Entries[0].Detail) {
+      request.params.Entries.length > 0) {
       try {
-        const details = JSON.parse(request.params.Entries[0].Detail)
-        details._datadog = {}
-        tracer.inject(span, 'text_map', details._datadog)
-        const finalData = JSON.stringify(details)
-        const byteSize = Buffer.byteLength(finalData)
-        if (byteSize >= (1024 * 256)) {
-          log.info('Payload size too large to pass context')
-          return
-        }
-        request.params.Entries[0].Detail = finalData
+        // eslint-disable-next-line no-console
+        console.log('AGOCS 1')
+        const currentMilliseconds = new Date().getMilliseconds()
+        request.params.Entries.forEach((entry, index) => {
+          // eslint-disable-next-line no-console
+          console.log('AGOCS 2')
+          const details = JSON.parse(entry.Detail)
+          details._datadog = {}
+          tracer.inject(span, 'text_map', details._datadog)
+          details._datadog.ms = currentMilliseconds
```

Comment:
This won't be extracted properly on the receiving end as it doesn't use the correct format. It would have to be added as a baggage item, although those are valid for the entire trace. Can you give some more context on what exactly this is doing? Maybe it could be handled differently without having to propagate this data?

Reply:
AWS EventBridge events are given a `time`, which by default is an RFC3339 of the time at which `PutEvents` was called at 1 second resolution. When a lambda function is triggered by an EventBridge event, we use the `time` as the start time of an inferred span. Since `time` only has 1 second resolution by default, it is possible for an EventBridge inferred span to start before its parent span. 

We are trying to include an optional field in the trace context that does not get propagated beyond the receiving thread of execution that includes the milliseconds portion of the timestamp, that we can use to start an inferred span at the correct time.

Reply:
Can you provide more information about how this work? We shouldn't be altering the result on trace injection and there might be a better way to do this. When you say thread of execution, that sounds like it's in the same process?